index.js

```diff
@@ -0,0 +1,292 @@
+
+let arrayOfNotes = [];
+
+function getSelectedValue() {
+let sortByVal = document.querySelector("#sortBy").value
+sortBy(sortByVal , arrayOfNotes)
+}
+
+
+
+
+// drawing functions
+function draw(arr){
+    console.log(arr)
+    clearBoard();
+    for (let index = 0; index < arr.length; index++) {
+      drawNote(arr[index]);
+    }
+
+}
+
+
+function clearBoard(){
+   NOTES_DOM.notes_board.innerHTML =  "";
+}
+function findIndex(arr,id) {
+  
+    for (let index = 0; index < arr.length; index++) {
+      if (arr[index].note_id == id) {
+
+        return index;
+      }
+    }
+}
+
+
+
+function  drawNote(stickyNote){
+   const { notes_board } = NOTES_DOM;
+   const currentNote = getNoteUi(stickyNote)
+   let currentid = currentNote.id.slice(0,10)
+    let currenTime = Date.now().toString().slice(0,10)
+    if (!currentNote) return;
+
+   if (currentid === currenTime){
+    $(currentNote).hide().appendTo(notes_board).fadeIn(1000);
+    
+   }
+   notes_board.append(currentNote);
+    
+}
+
+// main function save sticky note
+function saveStickyNOTE() {
+    
+    const {
+        note_name,
+        note_text,
+        note_date,
+        note_time
+    } = NOTES_DOM;
+
+  
+    
+
+// validate that all inputs are filled
+    if (!note_name.value || !note_text.value || !note_date.value || !note_time.value)  
+    {alert("Please Fill All Inputs") 
+    return;}
+    
+
+// regex validate that date and time is valid
+
+    const dateValidationResult = DateRegexValidate(note_date.value)
+    const timeValidationResult = timeRegexValidate(note_time.value)
+    if(!dateValidationResult){alert("Date is not valid") 
+    return;}  
+    if(!timeValidationResult){alert("Time is not valid") 
+    return;}  
+
+// validate that the due date and time is greatens than now
+
+  if( validateDate(note_date.value) === false || validateTime(note_time.value,note_date.value) === false ){
+      alert("Please enter date & time greatens than now") 
+    return;}
+
+let note_id = Date.now();
+let completed = false;
+let dateYear = note_date.value.slice(0, 4);
+let dateMonth = note_date.value.slice(5, 7);
+let dateDay = note_date.value.slice(8, 10);
+let note_dateInput = (dateDay + "-" + dateMonth + "-" + dateYear)
+
+    arrayOfNotes.push (new stickyNote(    
+        note_id,
+        note_name.value,
+        note_text.value,
+        note_dateInput,
+        note_time.value,
+        completed,
+    ));
+
+    saveToLocalStorage("stickyNotes",arrayOfNotes);
+    draw(arrayOfNotes)
+    NOTES_DOM.notes_board.reset();
+
+    }
+
+    // validation functions
+    function DateRegexValidate(date) {
+        let dateRegex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/
+        return dateRegex.test(date)
+    }
+    function timeRegexValidate(time) {
+        let timeRegex = /^(\d{1,2}):(\d{2})(?::(\d{2}))?$/
+        return timeRegex.test(time)
+    }
+    function validateTime(note_time,note_date) {
+        Date.prototype.toDateInputValue = (function() {
```

Comment:
When you add a function to a "class" prototype it is there forever.
You don't need to redefine it everytime. the definition of toDateInputValue in the bottom of this file is enough.

Remove all definitions of Date.prototype.toDateInputValue but the last one in the bottom of this file